File: 2021/Day02/solution.py
data = [[a, int(b)] for a,b in (line.split(' ') for line in open('./input.txt').readlines())]

def run_instruction(p2=False):
    h = d = a = 0

    for x in data:
        direction, num = x
        if not p2:
            if direction == 'forward':
                h += num
            if direction == 'down':
                d += num
            if direction == 'up':
                d -= num
        else:
            if direction == 'forward':
                h += num
                d += a * num
            if direction == 'down':
                a += num
            if direction == 'up':
                a -= num
        
    return h * d

print(run_instruction())
print(run_instruction(True))

File: 2021/Day02/solution.js
const fs = require('fs');
const data = fs.readFileSync('./input.txt', 'utf-8').split('\n');

const runInstructions = (p2 = false) => {
    let horizontal = 0;
    let depth = 0;
    let aim = 0; // Part 2

    for (const line of data) {
        const [dir, num] = line.split(' ');

        switch (dir) {
            case 'forward':
                if (p2) {
                    horizontal += parseInt(num);
                    depth += aim * parseInt(num);
                } else {
                    horizontal += parseInt(num);
                }
                break;
            case 'up':
                p2 ? aim -= parseInt(num) : depth -= parseInt(num);
                break;
            case 'down':
                p2 ? aim += parseInt(num) : depth += parseInt(num);
                break;
            default:
                break;
        }
    }

    return horizontal * depth
}




console.log(`Part 1: ${runInstructions()}`)
console.log(`Part 2: ${runInstructions(true)}`)
